### PR TITLE
Gradle instant project open

### PR DESCRIPTION
**Pull request by lkishalmi**

### CODE REVIEW

File: groovy/gradle/manifest.mf [hunk 1]
```diff
@@ -3,5 +3,5 @@ AutoUpdate-Show-In-Client: false
 OpenIDE-Module: org.netbeans.modules.gradle/2
 OpenIDE-Module-Layer: org/netbeans/modules/gradle/layer.xml
 OpenIDE-Module-Localizing-Bundle: org/netbeans/modules/gradle/Bundle.properties
-OpenIDE-Module-Specification-Version: 2.1
+OpenIDE-Module-Specification-Version: 2.1.1
```
Comment (matthiasblaesing) [inline groovy/gradle/manifest.mf]: For the integration into master I agree with this version, but for integration into release120 it should be 2.0.1.

**Review by matthiasblaesing — COMMENTED** [review]
Thank you - this looks good. These things got my attention:
1. The versions need special considerations. The version in this PR is fine for master, but should be 2.0.1 for release120.
2. I would consider squashing this PR into one big PR as that will make cherry-picking it into 2.0.1 easier.
3. This PR contains changes to the SPI/API, so there should be updates to the apichanges.xml (this ties to (1) as the version will be part of the xml (subminor 1)

File: groovy/gradle/manifest.mf (hunk 1 above)
Comment (lkishalmi) [reply groovy/gradle/manifest.mf]: Thank you for the review!
1. I'm aware of the different versions on master and on release120 branch and I'm going to do my merges according to that.
2. Agree squashing this for merge is planned.
3. API change documentation is coming soon, as we are fine with the other parts.

Thanks again!

File: groovy/gradle/src/org/netbeans/modules/gradle/ProjectTrust.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+import java.util.List;
+import java.util.Random;
+import java.util.prefs.Preferences;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import org.netbeans.api.project.Project;
+import org.openide.util.NbPreferences;
+
+/**
+ * This class allows Projects to store and check the trust property of for a project.
```
Comment (JaroslavTulach) [inline groovy/gradle/src/org/netbeans/modules/gradle/ProjectTrust.java]: "of for"

**Pull request closed without merging**